pkg/availability/multisigcollector/multisigcollector.go

```diff
@@ -48,3 +52,40 @@ func NewModule(mc *ModuleConfig, params *ModuleParams, nodeID t.NodeID) (modules
 
 	return m, nil
 }
+
+func NewReconfigurableModule(mc *ModuleConfig, nodeID t.NodeID, logger logging.Logger) modules.PassiveModule {
+	return factorymodule.New(
+		mc.Self,
+		factorymodule.DefaultParams(
+
+			// This function will be called whenever the factory module
+			// is asked to create a new instance of the multisig collector.
+			func(mscID t.ModuleID, params *factorymodulepb.GeneratorParams) (modules.PassiveModule, error) {
+
+				// Extract the IDs of the nodes in the membership associated with this instance
+				m := params.Type.(*factorymodulepb.GeneratorParams_MultisigCollector).MultisigCollector.Membership
+				mscNodeIDs := maputil.GetSortedKeys(t.Membership(m))
+
+				// Crate a copy of basic module config with an adapted ID for the submodule.
+				submc := *mc
+				submc.Self = mscID
+
+				// Create a new instance of the multisig collector.
+				multisigCollector, err := NewModule(
+					&submc,
+					&ModuleParams{
+						InstanceUID: []byte(mscID),
```

Comment:
The point of `InstanceUID` is to prevent signature replay attacks between instances of the protocol. It probably makes sense for the protocol using the multisig collector to also employ an instance UID and to use the concatenation of the "parent" `InstanceUID` with `mscID` as the `InstanceUID` for multisig collector.

However, it should probably be addressed in a separate PR. Could you at least put a TODO for it so that it doesn't get lost?